### ISSUE TITLE
Release 4.3.0 - Release candidate 6 - Manual tests - Deployment on Docker

### ISSUE THREAD

**Issue by vikman90** [wazuh/wazuh#13224]
The following issue aims to run the specified test for the current release candidate, report the results, and open new issues for any encountered errors.

## Test information

|Test name|Test|Deployment option|Main RC issue|Version|
|---|---|---|---|---|
|Deployment on Docker|Deployment on Docker|Docker|#13154|4.3.0 RC6|

## Test description

Test that the following proof of concept cases work in 4.3.0:

- Test rework of Docker repository
- Deployment on Docker - Follow this guide for 4.3.0 using Docker compose

## Test report procedure

All test results must have one of the following statuses: 
|                                  |                                            |
|---------------------------------|--------------------------------------------|
| :green_circle:  | All checks passed. |
| :red_circle:  | There is at least one failed check. |
| :yellow_circle:  | There is at least one expected failure or skipped test and no failures. |

Any failing test must be properly addressed with a new issue, detailing the error and the possible cause. 

An extended report of the test results must be attached as a ZIP or TXT file. Please attach any documents, screenshots or tables to the issue update with the results. This report can be used by the auditors to dig deeper into any possible failures and details.

## Conclusions

All tests have been executed and the results can be found [here](https://github.com/wazuh/wazuh/issues/13224#issuecomment-1109732194).

|Status|Test|Failure type|Notes|
|---|---|---|---|
|:red_circle: |https://github.com/wazuh/wazuh/issues/13224#issuecomment-1109732194|Error|https://github.com/wazuh/wazuh-docker/issues/622|
|:red_circle: |https://github.com/wazuh/wazuh/issues/13224#issuecomment-1109732194|Warning|https://github.com/wazuh/wazuh-docker/issues/621|
|:red_circle: |https://github.com/wazuh/wazuh/issues/13224#issuecomment-1109749141|Documentation|https://github.com/wazuh/wazuh-documentation/issues/5115|
|:red_circle: |https://github.com/wazuh/wazuh/issues/13224#issuecomment-1109732194|Documentation|https://github.com/wazuh/wazuh-documentation/issues/5113|
|:red_circle: |https://github.com/wazuh/wazuh/issues/13224#issuecomment-1109732194|MacOS functional|https://github.com/wazuh/wazuh-docker/issues/619|
|:red_circle: |https://github.com/wazuh/wazuh/issues/13224#issuecomment-1109732194|Documentation|https://github.com/wazuh/wazuh-documentation/issues/5110|
|:red_circle: |https://github.com/wazuh/wazuh/issues/13224#issuecomment-1109749141|Documentation|https://github.com/wazuh/wazuh-documentation/issues/5114|
|:red_circle: |https://github.com/wazuh/wazuh/issues/13224#issuecomment-1109732194|Documentation|https://github.com/wazuh/wazuh-documentation/issues/5111|
|:red_circle: |https://github.com/wazuh/wazuh/issues/13224#issuecomment-1109732194|Documentation|https://github.com/wazuh/wazuh-documentation/issues/5112|

All tests have passed and the fails have been reported or justified. Therefore, I conclude that this issue is finished and OK for this release candidate.

## Auditors' validation
The definition of done for this one is the validation of the conclusions and the test results from all auditors.

All checks from below must be accepted in order to close this issue.

- [x] @alberpilot

**Comment by jnasselle** [comment]
### Apr 25, 2022

### Environment
Kernel: Linux t14 5.10.105-1-MANJARO #1 SMP PREEMPT Fri Mar 11 14:12:33 UTC 2022 x86_64 GNU/Linux
OS: Manjaro Linux
Specs: Ryzen 4750U, 16GB RAM
**Goal:** check the overall installation process and validate product functionalities work as expected 

## Conclusions so far
- There's no mention of local docker images building procedure, only fetching them from dockerhub by default
- Requirements are widely open when talking about architecture: `64 bits`. It was proven that only amd64 is currently working OOTB
- There are no clues about Wazuh Dashboard access credentials after deployment. Adding this will enhance UX, mostly in a first-approach scenario.

## Tests
Preconditions
- Changes were made in order to use no-yet released dockerhub images, tagged as `4.3.0-dev` rather than `4.3.0` (expected by docker compose). Applyig the next patch in single-node and multi-node should  be enough
```patch
diff --git a/single-node/docker-compose.yml b/single-node/docker-compose.yml
index 827421c..7278b28 100644
--- a/single-node/docker-compose.yml
+++ b/single-node/docker-compose.yml
@@ -3,7 +3,7 @@ version: '3.7'
 
 services:
   wazuh.manager:
-    image: wazuh/wazuh-manager:4.3.0
+    image: wazuh/wazuh-manager:4.3.0-dev
     hostname: wazuh.manager
     restart: always
     ports:
@@ -39,7 +39,7 @@ services:
       - ./config/wazuh_cluster/wazuh_manager.conf:/wazuh-config-mount/etc/ossec.conf
 
   wazuh.indexer:
-    image: wazuh/wazuh-indexer:4.3.0
+    image: wazuh/wazuh-indexer:4.3.0-dev
     hostname: wazuh.indexer
     restart: always
     ports:
@@ -64,7 +64,7 @@ services:
       - ./config/wazuh_indexer/internal_users.yml:/usr/share/wazuh-indexer/plugins/opensearch-security/securityconfig/internal_users.yml
 
   wazuh.dashboard:
-    image: wazuh/wazuh-dashboard:4.3.0
+    image: wazuh/wazuh-dashboard:4.3.0-dev
     hostname: wazuh.dashboard
     restart: always
     ports:
```
- In case user want to build docker images locally, Some Dockerfiles should be changed in order to retrieve not-yet released packages
```patch
diff --git a/build-docker-images/wazuh-dashboard/Dockerfile b/build-docker-images/wazuh-dashboard/Dockerfile
index 3853f18..8b131f6 100644
--- a/build-docker-images/wazuh-dashboard/Dockerfile
+++ b/build-docker-images/wazuh-dashboard/Dockerfile
@@ -25,7 +25,7 @@ RUN $INSTALL_DIR/bin/opensearch-dashboards-keystore create --allow-root && \
     echo kibanaserver | $INSTALL_DIR/bin/opensearch-dashboards-keystore add opensearch.password --stdin --allow-root
 
 # Install Wazuh App
-RUN $INSTALL_DIR/bin/opensearch-dashboards-plugin install https://packages.wazuh.com/4.x/ui/dashboard/wazuh-${WAZUH_VERSION}.zip --allow-root
+RUN $INSTALL_DIR/bin/opensearch-dashboards-plugin install https://packages-dev.wazuh.com/pre-release/ui/dashboard/wazuh-${WAZUH_VERSION}.zip --allow-root
 
 # Copy and set permissions to config files
 COPY config/opensearch_dashboards.yml $INSTALL_DIR/config/
diff --git a/build-docker-images/wazuh-manager/Dockerfile b/build-docker-images/wazuh-manager/Dockerfile
index 11a852e..0c41f3f 100644
--- a/build-docker-images/wazuh-manager/Dockerfile
+++ b/build-docker-images/wazuh-manager/Dockerfile
@@ -10,7 +10,7 @@ ARG WAZUH_FILEBEAT_MODULE="wazuh-filebeat-0.1.tar.gz"
 RUN apt-get update && apt install curl apt-transport-https lsb-release gnupg -y
 
 RUN apt-key adv --fetch-keys https://packages.wazuh.com/key/GPG-KEY-WAZUH && \
-    echo "deb https://packages.wazuh.com/4.x/apt/ stable main" | tee -a /etc/apt/sources.list.d/wazuh.list && \
+    echo "deb https://packages-dev.wazuh.com/pre-release/apt/ unstable main" | tee -a /etc/apt/sources.list.d/wazuh.list && \
     apt-get update && \
     apt-get install wazuh-manager=${WAZUH_VERSION}-1
 
@@ -48,4 +48,4 @@ RUN chmod 755 /permanent_data.sh && \
 # Services ports
 EXPOSE 55000/tcp 1514/tcp 1515/tcp 514/udp 1516/tcp
```


### Single node deployment

:green_circle: Cloning and checkout branch
:yellow_circle:  Certificate creation
![Screenshot from 2022-04-25 12-51-44](https://user-images.githubusercontent.com/1791430/165295109-b7b7a4c8-351a-485b-906a-71afb7c9941d.png)
![Screenshot from 2022-04-25 12-54-20](https://user-images.githubusercontent.com/1791430/165296337-5215f1cb-a003-4fd7-86b6-b3afeda210da.png)
**Certificates permission should be checked**
:green_circle: Set wm_map_count value
![Screenshot from 2022-04-25 12-52-42](https://user-images.githubusercontent.com/1791430/165296579-874a82ef-4dc6-4ffa-83b8-3618d996c239.png)
:yellow_circle: Start deployment (foreground)
![Screenshot from 2022-04-25 14-00-12](https://user-images.githubusercontent.com/1791430/165298559-20e36c1c-ae84-418c-9d18-3d3ad69e5912.png)
![Screenshot from 2022-04-25 13-59-47](https://user-images.githubusercontent.com/1791430/165298569-357d7588-ed56-41a7-8bfc-ccd12c1dd4eb.png)
Deployment seems to be OK but we need check that warns
:yellow_circle: Wazuh Dashboard access 
![Screenshot from 2022-04-25 14-35-31](https://user-images.githubusercontent.com/1791430/165298855-d6b54007-4426-43bb-96bb-e5bec5585d69.png)
Guide ends here, giving no default credentials to access Wazuh Dashboard. Despite user customization is explained here, IMHO is not enough.

**Comment by RamiroRD** [comment]
# Deployment on Docker (AMD64, Ubuntu)

## Docker installation :yellow_circle: 

<details>
<summary>docker installation</summary>

1. Architecture and kernel version check :green_circle: 

```
vagrant@wazuh-manager:~/wazuh/src$ uname -a
Linux wazuh-manager 5.4.0-109-generic #123-Ubuntu SMP Fri Apr 8 09:10:54 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux
```

2. Install docker-engine. :green_circle:  

```
vagrant@wazuh-manager:~/wazuh/src$ curl -sSL https://get.docker.com/ | sh
# Executing docker install script, commit: 0221adedb4bcde0f3d18bddda023544fc56c29d1
+ sudo -E sh -c apt-get update -qq >/dev/null
+ sudo -E sh -c DEBIAN_FRONTEND=noninteractive apt-get install -y -qq apt-transport-https ca-certificates curl >/dev/null
+ sudo -E sh -c curl -fsSL "https://download.docker.com/linux/ubuntu/gpg" | gpg --dearmor --yes -o /usr/share/keyrings/docker-archive-keyring.gpg
gpg: WARNING: unsafe ownership on homedir '/home/vagrant/.gnupg'
+ sudo -E sh -c chmod a+r /usr/share/keyrings/docker-archive-keyring.gpg
+ sudo -E sh -c echo "deb [arch=amd64 signed-by=/usr/share/keyrings/docker-archive-keyring.gpg] https://download.docker.com/linux/ubuntu focal stable" > /etc/apt/sources.list.d/docker.list
+ sudo -E sh -c apt-get update -qq >/dev/null
+ sudo -E sh -c DEBIAN_FRONTEND=noninteractive apt-get install -y -qq --no-install-recommends docker-ce docker-ce-cli docker-compose-plugin docker-scan-plugin >/dev/null
+ version_gte 20.10
+ [ -z  ]
+ return 0
+ sudo -E sh -c DEBIAN_FRONTEND=noninteractive apt-get install -y -qq docker-ce-rootless-extras >/dev/null

================================================================================

To run Docker as a non-privileged user, consider setting up the
Docker daemon in rootless mode for your user:

    dockerd-rootless-setuptool.sh install

Visit https://docs.docker.com/go/rootless/ to learn about rootless mode.


To run the Docker daemon as a fully privileged service, but granting non-root
users access, refer to https://docs.docker.com/go/daemon-access/

WARNING: Access to the remote API on a privileged Docker daemon is equivalent
         to root access on the host. Refer to the 'Docker daemon attack surface'
         documentation for details: https://docs.docker.com/go/attack-surface/

================================================================================


```

**NOTE: The official docker documentation doesn't talk about docker-engine installation for Amazon Linux 2. We decided to use the official package delivered by the OS maintainers. The packages are only available for Amazon Linux 2 not 1.**

3. Start the Docker service :green_circle: 
 
```
vagrant@wazuh-manager:~/wazuh/src$ sudo systemctl start docker
vagrant@wazuh-manager:~/wazuh/src$ sudo systemctl status docker
● docker.service - Docker Application Container Engine
     Loaded: loaded (/lib/systemd/system/docker.service; enabled; vendor preset: enabled)
     Active: active (running) since Tue 2022-04-26 12:22:26 UTC; 1 day 5h ago
TriggeredBy: ● docker.socket
       Docs: https://docs.docker.com
   Main PID: 945 (dockerd)
      Tasks: 10
     Memory: 35.9M
     CGroup: /system.slice/docker.service
             └─945 /usr/bin/dockerd -H fd:// --containerd=/run/containerd/containerd.sock

Apr 26 12:22:25 wazuh-manager dockerd[945]: time="2022-04-26T12:22:25.910621083Z" level=warning msg="Your kernel does not support CPU realtime sc>
Apr 26 12:22:25 wazuh-manager dockerd[945]: time="2022-04-26T12:22:25.910626894Z" level=warning msg="Your kernel does not support cgroup blkio we>
Apr 26 12:22:25 wazuh-manager dockerd[945]: time="2022-04-26T12:22:25.910630987Z" level=warning msg="Your kernel does not support cgroup blkio we>
Apr 26 12:22:25 wazuh-manager dockerd[945]: time="2022-04-26T12:22:25.912375302Z" level=info msg="Loading containers: start."
Apr 26 12:22:26 wazuh-manager dockerd[945]: time="2022-04-26T12:22:26.417783460Z" level=info msg="Default bridge (docker0) is assigned with an IP>
Apr 26 12:22:26 wazuh-manager dockerd[945]: time="2022-04-26T12:22:26.551046208Z" level=info msg="Loading containers: done."
Apr 26 12:22:26 wazuh-manager dockerd[945]: time="2022-04-26T12:22:26.677978222Z" level=info msg="Docker daemon" commit=87a90dc graphdriver(s)=ov>
Apr 26 12:22:26 wazuh-manager dockerd[945]: time="2022-04-26T12:22:26.681201485Z" level=info msg="Daemon has completed initialization"
Apr 26 12:22:26 wazuh-manager systemd[1]: Started Docker Application Container Engine.
Apr 26 12:22:26 wazuh-manager dockerd[945]: time="2022-04-26T12:22:26.743044181Z" level=info msg="API listen on /run/docker.sock"
```


</details>

## Docker compose  :green_circle: 

<details>

<summary>docker compose</summary>

1. Download the Docker Compose binary :green_circle: 

```
vagrant@wazuh-manager:~/wazuh/src$ sudo curl -L "https://github.com/docker/compose/releases/download/1.29.2/docker-compose-$(uname -s)-$(uname -m)" -o /usr/local/bin/docker-compose
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   664  100   664    0     0   4397      0 --:--:-- --:--:-- --:--:--  4397
100 12.1M  100 12.1M    0     0  16.5M      0 --:--:-- --:--:-- --:--:-- 33.0M
```

2. Grant execution permissions :green_circle: 
```
vagrant@wazuh-manager:~/wazuh/src$ sudo chmod +x /usr/local/bin/docker-compose
vagrant@wazuh-manager:~/wazuh/src$
```

3. Test the installation to ensure everything went properly :green_circle: 
```
vagrant@wazuh-manager:~/wazuh/src$ docker-compose version
docker-compose version 1.29.2, build 5becea4c
docker-py version: 5.0.0
CPython version: 3.7.10
OpenSSL version: OpenSSL 1.1.0l  10 Sep 2019
```

</details>


## Wazuh Docker deployment :red_circle: 

<details>

<summary>Wazuh Docker deployment</summary>

### Requirements:red_circle:  

<details>

<summary>requirements</summary>

1. Increase max_map_count on your host :green_circle:  

```
vagrant@wazuh-manager:~/wazuh-docker/single-node/config$ sysctl vm.max_map_count
vm.max_map_count = 262144
```

### Usage. Single Node :green_circle:

<details>

<summary>single-node</summary>

1. Clone the Wazuh repository to your system :green_circle: 

```
vagrant@wazuh-manager:~$ git clone https://github.com/wazuh/wazuh-docker/ -b 4.3 --depth 1 
Cloning into 'wazuh-docker'...
remote: Enumerating objects: 90, done.
remote: Counting objects: 100% (90/90), done.
remote: Compressing objects: 100% (79/79), done.
remote: Total 90 (delta 16), reused 33 (delta 5), pack-reused 0
Unpacking objects: 100% (90/90), 47.24 KiB | 1.21 MiB/s, done.
```

2. Generate certificate 🔴 :

3. Start Wazuh using docker-compose :red_circle:

The path of the yml file used for generating the certificates is wrong on the documentation:
![Screen Shot 2022-04-27 at 16 24 17](https://user-images.githubusercontent.com/6182640/165604258-c9df0fbc-32ea-4ec6-953b-1f3b78dd9443.png)
While this is the real path:
```
vagrant@wazuh-manager:~/wazuh-docker/single-node/config$ ls
certs.yml  wazuh_cluster  wazuh_dashboard  wazuh_indexer
vagrant@wazuh-manager:~/wazuh-docker/single-node/config$ cat certs.yml 
nodes:
  # Wazuh indexer server nodes
  indexer:
    - name: wazuh.indexer
      ip: wazuh.indexer

  # Wazuh server nodes
  # Use node_type only with more than one Wazuh manager
  server:
    - name: wazuh.manager
      ip: wazuh.manager

  # Wazuh dashboard node
  dashboard:
    - name: wazuh.dashboard
```
</details>
This was apparently changed [here](https://github.com/wazuh/wazuh-docker/commit/c4e2cc54c50573186821f369e3a386a1d68943ff).

After thinking editing `certs.yml`, I realized it's not just a documentation issue, the generation process fails altogether:
```
vagrant@wazuh-manager:~/wazuh-docker/single-node$ docker-compose -f generate-indexer-certs.yml run --rm generator
Creating single-node_generator_run ... done
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 24995  100 24995    0     0  38993      0 --:--:-- --:--:-- --:--:-- 38993
Cert tool exists in Packages-dev bucket
cp: cannot stat '/certificates/certs.yml': No such file or directory
27/04/2022 19:51:48 ERROR: No configuration file found.
ERROR: 1
```

### Usage. Multi Node 🔴

<details>

<summary>multi-node</summary>

Blocked until I get single-node to work.
</details>

</details>
